hooks/useRailgunProvider.tsx

```diff
@@ -1,19 +1,43 @@
 import { useEffect, useState } from 'react';
 import { setProviderForNetwork } from '@railgun-community/quickstart';
+import { BigNumber } from 'ethers';
 import { useNetwork, useProvider } from 'wagmi';
 import { networks } from '@/utils/networks';
 import { loadProviders } from '@/utils/railgun';
 
+// Fee is in bips, e.g. a value of 25 is a 0.25% fee.
+interface ShieldFee {
+  [chainId: number]: BigNumber;
+}
+
+const initialShieldFees: ShieldFee = {};
+Object.keys(networks).forEach((chainId) => {
+  // Current fees are 0.25% everywhere, so we initialize with that as a default -- If something goes
+  // wrong fetching fees, better to show a higher fee than a zero default fee.
+  initialShieldFees[Number(chainId)] = BigNumber.from('25');
+});
+
 export const useRailgunProvider = () => {
   const [isProviderLoaded, setProviderLoaded] = useState<Boolean>(false);
+  const [shieldFees, setShieldFees] = useState<ShieldFee>(initialShieldFees);
   const { chain } = useNetwork();
   const network = networks[chain?.id || 1];
   const provider = useProvider();
 
   useEffect(() => {
     const fn = async () => {
       setProviderLoaded(false);
-      await loadProviders();
+      const res = await loadProviders();
+
+      // Set the shield fees for each network. For convenience, we just update the
+      // `initialShieldFees` object then set it.
+      res.forEach((r) => {
+        const newFee = r.providerInfo.feesSerialized?.shield;
+        initialShieldFees[r.chainId] = BigNumber.from(newFee || initialShieldFees[r.chainId]);
+      });
+      setShieldFees(initialShieldFees);
+
+      // Provider is done loading.
```

Comment:
The one improvement we could make here is to decouple this logic from loading the providers, mainly because anywhere this hook is called the providers will be reloaded. The cleanest way to do this may be to some of this to context, but definitely a future improvement.

Reply:
Yea, I was also thinking of wrapping it within `useSWR` but wasn't sure if it's bad to use `useSWR` within a `useEffect` hook?

Reply:
Ah just tried, doesn't really fit here since we have the `setProviderLoaded` side effect. Looks like it runs 4 times and then never runs again (even when switching networks), so this probably ok for now

Reply:
Now tracking in https://github.com/ScopeLift/railgun-shielder/issues/80

Reply:
I don't think we should call hooks inside useEffect because it breaks the [rules of hooks](https://reactjs.org/docs/hooks-rules.html).  We could move load providers to the top level and wrap in an SWR hook and I guess it would be cached if it is called multiple times. For network switching did you add the chain id as a value for the cache key? My guess is that was why it didn't update.

Reply:
> I don't think we should call hooks inside useEffect because it breaks the [rules of hooks](https://reactjs.org/docs/hooks-rules.html).

Right that doesn't work. To clarify I tried removing `useEffect` and putting the side effect within the `useSWR` fetcher. Sorry, I wasn't clear on that

> For network switching did you add the chain id as a value for the cache key? My guess is that was why it didn't update.

What didn't update on network switching? To clarify, right now `loadProviders` fetches the info for all supported chains, so it only _needs_ to run once (or poll if we think the fee could change during the time a user is using the app, but not worried about that for now). What I meant to convey is that right now, the current implementation of the PR, has the current hook run 4 times (some re-renders occur after initial load I guess), and that's it. So since it's not constantly running it seems ok for now even if not technically optimal

My message was very unclear, sorry 😅

Reply:
Ah okay, thank you for clarifying!